src/main/java/com/microsoft/sqlserver/jdbc/ISQLServerMessageHandler.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Microsoft JDBC Driver for SQL Server Copyright(c) Microsoft Corporation All rights reserved. This program is made
+ * available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+package com.microsoft.sqlserver.jdbc;
+
+public interface ISQLServerMessageHandler
+{
+    /**
+     * You can use the ISQLServerMessageHandler interface to customize the way JDBC handles error messages generated by the SQL Server. 
+     * Implementing ISQLServerMessageHandler in your own class for handling error messages can provide the following benefits:
+     * <ul>
+     *   <li><b>"message feedback"</b><br>
+     *       Display Server messages from a long running SQL Statement<br>
+     *       Like <code>RAISERROR ('Progress message...', 0, 1) WITH NOWAIT</code><br>
+     *       Or Status messages from a running backup...<br>
+     *   </li>
+     *   <li><b>"Universal" error logging</b><br>
+     *       Your error-message handler can contain the logic for handling all error logging.
+     *   </li>
+     *   <li><b>"Universal" error handling</b><br>
+     *       Error-handling logic can be placed in your error-message handler, instead of being repeated throughout your application.
+     *   </li>
+     *   <li><b>Remapping of error-message severity</b>, based on application requirements<br>
+     *       Your error-message handler can contain logic for recognizing specific error messages, and downgrading or upgrading 
+     *       their severity based on application considerations rather than the severity rating of the server. 
+     *       For example, during a cleanup operation that deletes old rows, you might want to downgrade the severity of a 
+     *       message that a row does not exist. However, you may want to upgrade the severity in other circumstances.
+     *   </li>
+     * </ul>
+     * 
+     * Example code:
+     * <pre>
+     *  public ISQLServerMessage messageHandler(ISQLServerMessage srvErrorOrWarning)

```

Comment:
nitpick: since this is a public API, can you please use a full spelling i.e. serverErrorOrWarning